Jellyfin.Server/Migrations/Routines/MigrateLibraryDb.cs

```diff
@@ -74,265 +74,306 @@ public void Perform()
-        {
-            var chapter = GetChapter(dto);
-            dbContext.Chapters.Add(chapter);
+            using (new TrackedMigrationStep($"saving {operation.JellyfinDbContext.Chapters.Local.Count} Chapters entries", _logger))
+            {
+                operation.JellyfinDbContext.SaveChanges();
+            }
         }
 
-        _logger.LogInformation("Try saving {0} Chapters entries.", dbContext.Chapters.Local.Count);
-        dbContext.SaveChanges();
-        migrationTotalTime += stopwatch.Elapsed;
-        _logger.LogInformation("Saving Chapters took {0}.", stopwatch.Elapsed);
-        stopwatch.Restart();
-
-        _logger.LogInformation("Start moving AncestorIds.");
-        const string ancestorIdsQuery = """
-        SELECT ItemId, AncestorId, AncestorIdText FROM AncestorIds
-        WHERE
-        EXISTS(SELECT 1 FROM TypedBaseItems WHERE TypedBaseItems.guid = AncestorIds.ItemId)
-        AND
-        EXISTS(SELECT 1 FROM TypedBaseItems WHERE TypedBaseItems.guid = AncestorIds.AncestorId)
-        """;
-        dbContext.AncestorIds.ExecuteDelete();
-
-        foreach (SqliteDataReader dto in connection.Query(ancestorIdsQuery))
+        using (var operation = GetPreparedDbContext("moving AncestorIds"))
         {
-            var ancestorId = GetAncestorId(dto);
-            dbContext.AncestorIds.Add(ancestorId);
-        }
+            const string ancestorIdsQuery =
+            """
+            SELECT ItemId, AncestorId, AncestorIdText FROM AncestorIds
+            WHERE
+            EXISTS(SELECT 1 FROM TypedBaseItems WHERE TypedBaseItems.guid = AncestorIds.ItemId)
+            AND
+            EXISTS(SELECT 1 FROM TypedBaseItems WHERE TypedBaseItems.guid = AncestorIds.AncestorId)
+            """;
 
-        _logger.LogInformation("Try saving {0} AncestorIds entries.", dbContext.AncestorIds.Local.Count);
+            using (new TrackedMigrationStep("loading AncestorIds", _logger))
+            {
+                foreach (SqliteDataReader dto in connection.Query(ancestorIdsQuery))
+                {
+                    var ancestorId = GetAncestorId(dto);
+                    operation.JellyfinDbContext.AncestorIds.Add(ancestorId);
+                }
+            }
 
-        dbContext.SaveChanges();
-        migrationTotalTime += stopwatch.Elapsed;
-        _logger.LogInformation("Saving AncestorIds took {0}.", stopwatch.Elapsed);
-        stopwatch.Restart();
+            using (new TrackedMigrationStep($"saving {operation.JellyfinDbContext.AncestorIds.Local.Count} AncestorId entries", _logger))
+            {
+                operation.JellyfinDbContext.SaveChanges();
+            }
+        }
 
         connection.Close();
+
         _logger.LogInformation("Migration of the Library.db done.");
         _logger.LogInformation("Move {0} to {1}.", libraryDbPath, libraryDbPath + ".old");
+        _logger.LogInformation("Migrating Library db took {0}.", fullOperationTimer.Elapsed);
```

Comment:
```suggestion
        File.Move(libraryDbPath, libraryDbPath + ".old", true);
        _logger.LogInformation("Migrating Library db took {0}.", fullOperationTimer.Elapsed);
```

Don't forget to remove the File.Move from below.